src/plugins/home/public/application/components/solutions_section/solution_panel.tsx

```diff
@@ -62,25 +62,24 @@ export const SolutionPanel: FC<Props> = ({ solution, apps = [] }) =>
   solution && apps.length ? (
     <EuiFlexItem
       key={solution.id}
-      className={`homSolutionsSection__container ${
-        solution.id === 'kibana' ? 'homSolutionsSection__single' : ''
-      } ${solution.className}`}
+      className={`homSolutions__item ${solution.id === 'kibana' ? 'homSolutions__group' : ''}`}
       grow={1}
     >
       <EuiPanel
         paddingSize="none"
-        className="homSolutionsSection__panel"
+        className={`homSolutionPanel  homSolutionPanel--${solution.id}`}
```

Comment:
@cqliu1 should clarify, but there was feedback on the dev side that we should not hard code solution id's. Instead, the home page should remain agnostic while the plugins themselves push content into the home page.

As seen below, this results in the plugin registering a CSS class name which is unconventional, but allows us to push some complexity down the roadmap. In other words, we know this could be improved but we don't have a current need (nor do we anticipate any in the near team) for additional solutions to be added. At the point another solution or further content is needed, then we could improve upon the plugin registration format (i.e. use an enumerated prop instead of a class name).

Reply:
Yeah, @cqliu1 and I spoke about the hard-coding of classes issue. I thought it was more the issue of hard-coding the solution classes (versus dynamically generating the classes, as I'm doing here by concatenating the `solution.id` prop to a class prefix string) was the issue. But perhaps I misunderstood the issue? If I have, then I'm not sure why classifying based on the content type (in this case, solution name) would be frowned upon if that is the case. Would it be possible to elaborate, assuming I've misunderstood?

Reply:
Since all the home plugin code lives in OSS, we want to avoid hardcoding names or other things that refer to licensed features, which I would assume includes CSS selectors as well.

Reply:
Forgive me, but I'm still struggling to understand why this approach would be frowned upon. For the OSS situation, since the solution panels wouldn't be rendered on the home page, wouldn't these classes be absent from the OSS version of Kibana? Or do you mean simply having those classes present in the SCSS/CSS is problematic (regardless if the elements with these classes are rendered)?

Reply:
If I understand correctly, OSS code is rendering an element with a CSS class that is derived from the name of a solution that is registered via X-Pack. Is the CSS that applies to these CSS classes also in X-pack? If so, that seems reasonable. The only downside is that X-Pack will need to know the details of how these CSS classes are derived from the name of the solution, and this is somewhat brittle.

Reply:
The CSS styles that apply to these classes live in OSS. Michael's changes generate these selectors with the `id` registered by each solution, so no IDs are hard coded in the `Home` component, but these classes with solution IDs are hardcoded into `_solutions_panel.scss` which lives in the Home plugin.

Reply:
Gotcha... We've already let the concept of "solutions" creep into OSS with https://github.com/elastic/kibana/blob/f7b5144e1d991e1b679dbeed168a41648c571dd0/src/core/utils/default_app_categories.ts#L29 and I thought you were going to take advantage of that elsewhere @cqliu1 in OSS for the hom-page. Were you able to find a way around this @cqliu1 or are you primarily concerned with allowing even more X-Pack concerns from being implemented in OSS?

Reply:
I ended up not needing to use `DEFAULT_APP_CATEGORIES` and found a way to refactor all of the home page redesign code without leaving any reference to features unavailable in OSS. These CSS classes are the only place where these solution IDs would appear. I was trying to avoid any mention of specific features/solutions in the home plugin code, but I don't feel that strongly against it if it's not a big deal to have solution specific selectors in Home plugin CSS files.

Reply:
>  I ended up not needing to use DEFAULT_APP_CATEGORIES and found a way to refactor all of the home page redesign code without leaving any reference to features unavailable in OSS

That's great, and even better than I thought!! Given that, I think it makes sense to explore alternatives to hard-coding these solution specific styles in OSS. Ideally, OSS wouldn't know anything about X-Pack specific concerns. This ensures that we're creating OSS APIs in a truly extensible manner.

Reply:
Just to make sure I understand, even though the only place that these solution-specific modifier classes would appear in OSS version of Kibana is the one CSS file, we still wish to change them to no longer reference the solutions? 

If so, anyone have any recommendations that aren't solution or style oriented?

Reply:
I think this may circle back to what was previously in place and started this thread.

Say we have generic named classes that define several alternatives for card 'branding'. With those in place, the solution can define which brand it wants to follow. I realize this is all a little artificial feeling (although the solutions do have logos that we can lean on for color selection).

We could, in the future, expand this concept so that each solution plugin defines their brand settings in the plugin (register a brand color, namely). Catherine and I discussed this, but decided it was a bit much to bite off at the moment which is how we landed on the somewhat clunky class name approach. If it feels better, we could have each solution pass a brand color that is used for the header background color. In that case though, we might then need to do without the background images as that starts to get more complex and messy for this first version.

Reply:
I'm not sure I've got a good alternative... I'll keep brainstorming, but let's go ahead and merge in these changes and leave the solution specific CSS selectors in there for now.